### PR TITLE
[Quality][tidy] Add missing headers to MIOpen_source

### PR DESCRIPTION
**Pull request by atamazov**
Review only the 2nd commit - 96bc894860ce2211e73e25d891843c2a09758b17. The 1st one is simple sorting.

This affects tidy checks.

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -72,200 +72,288 @@ function(add_kernels FILE_NAME VAR_PREFIX VAR_SUFFIX KERNEL_FILES)
...
+    include/miopen/object.hpp
+    include/miopen/ocldeviceinfo.hpp
     include/miopen/oclkernel.hpp
-    include/miopen/tensor.hpp
-    include/miopen/tensor_layout.hpp
-    include/miopen/tensor_ops.hpp
+    include/miopen/op_kernel_args.hpp
+    include/miopen/par_for.hpp
+    include/miopen/perf_field.hpp
+    include/miopen/pooling
```
Comment (junliume) [inline src/CMakeLists.txt:211]:
```suggestion
```

Is this line accidentally added? :-P

Comment (atamazov) [reply src/CMakeLists.txt:211]: :/

**Comment by atamazov** [comment]
@pfultz2 Could you please review this? I do not see any merits from adding headers except more work for clang-tidy.

**Comment by pfultz2** [comment]
> I do not see any merits from adding headers except more work for clang-tidy.

I dont see the merits for this either(as the headers are checked when included by .cpp files). However, I dont see any issue with merging this in either.

**Review by pfultz2 — APPROVED** [review]

**Comment by atamazov** [comment]
@pfultz2 maybe it is better to remove headers totally instead of this?

**Comment by atamazov** [comment]
On hold until next promotion to Staging branch

**Comment by atamazov** [comment]
> @pfultz2 maybe it is better to remove headers totally instead of this?

:ear:

**Review by DrizztDoUrden — APPROVED** [review]
LGTM

**Comment by atamazov** [comment]
@DrizztDoUrden Thanks, but --

> > @pfultz2 maybe it is better to remove headers totally (_from CMakeFile_) instead of this?

What do you think?

**Comment by pfultz2** [comment]
Yea that might be better as I don't really understand the reason to add them in the first place.

**Comment by JehandadKhan** [comment]
> Yea that might be better as I don't really understand the reason to add them in the first place.

+1

**Comment by junliume** [comment]
> > Yea that might be better as I don't really understand the reason to add them in the first place.
> 
> +1

Recommendation:
Let's merge this PR as it is an enhancement of current practice. Create an issue for removing all header files from tidy checks (they will be checked as parts of the files that include them anyway), and move on with resolving that issue later. :)

**Review by junliume — APPROVED** [review]

**Comment by junliume** [comment]
Created https://github.com/ROCmSoftwarePlatform/MIOpen/issues/1365

**Pull request merged**